<a href="https://colab.research.google.com/github/felixzhao/STAT501-Assignment/blob/main/notebooks/T4_FZ_correlation.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import os
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
q2_data_path = "/content/drive/My Drive/Study@NZ/VUW@NZ/STATS 501/assignment/data/q2.csv"
q3_data_path = "/content/drive/My Drive/Study@NZ/VUW@NZ/STATS 501/assignment/data/q3.csv"
q4_data_path = "/content/drive/My Drive/Study@NZ/VUW@NZ/STATS 501/assignment/data/q4.csv"

In [4]:
q2_cols = ["Income", "Support"]
q3_cols = ["HDI", "Interest"]
q4_cols = ["Support", "Interest", "Overall"]

In [5]:
q2_df = pd.read_csv(q2_data_path)[q2_cols]
q3_df = pd.read_csv(q3_data_path)[q3_cols]
q4_df = pd.read_csv(q4_data_path)[q4_cols]

# Question 2

In [7]:
# Calculate Pearson's Correlation
correlation_matrix = q2_df.corr()
print("Pearson's Correlation between income and support: ", correlation_matrix.loc['Income', 'Support'])


Pearson's Correlation between income and support:  -0.4772964481261777


# Question 3

In [9]:
# Calculate Pearson's Correlation
correlation_matrix = q3_df.corr()
print("Pearson's Correlation between HDI and Interest: ", correlation_matrix.loc['HDI', 'Interest'])

Pearson's Correlation between HDI and Interest:  -0.8196259950347936


# Question 4

In [10]:
# Generate a correlation matrix
correlation_matrix = q4_df[['Support', 'Interest', 'Overall']].corr()

# Display the correlation matrix
print(correlation_matrix)

           Support  Interest   Overall
Support   1.000000  0.795298 -0.509036
Interest  0.795298  1.000000 -0.784566
Overall  -0.509036 -0.784566  1.000000


## Interpretation of Individual Correlation Coefficients

1. Support and Interest (0.795298): This positive correlation suggests that as support levels increase, interest levels also tend to increase, and vice versa. The value of approximately 0.80 is fairly strong, suggesting a substantial positive linear relationship between the two.

2. Support and Overall (-0.509036): The negative correlation suggests that as support levels increase, the "Overall" value tends to decrease, and vice versa. The value of approximately -0.51 suggests a moderate negative linear relationship.

3. Interest and Overall (-0.784566): This negative correlation suggests that as interest levels increase, the "Overall" value tends to decrease, and vice versa. The value of approximately -0.78 indicates a strong negative linear relationship.